cypress/integration/core-opensearch-dashboards/opensearch-dashboards/apps/query_enhancements/field_display_filtering.spec.js

```diff
@@ -0,0 +1,245 @@
+
+const randomString = Math.random().toString(36).substring(7);
+const workspace = `${WORKSPACE_NAME}-${randomString}`;
+
+const selectDataset = (datasetType, language) => {
+  switch (datasetType) {
+    case 'index':
+      dataExplorer.selectIndexDataset(DATASOURCE_NAME, INDEX_NAME, language);
+      break;
+    case 'index_pattern':
+      dataExplorer.selectIndexPatternDataset(INDEX_PATTERN_NAME, language);
+      break;
+  }
+};
+const setDateRange = (datasetType, language) => {
+  if (language === 'OpenSearch SQL') {
+    return;
+  } else {
+    cy.setTopNavDate(START_TIME, END_TIME);
+  }
+};
+const verifyTableFieldFilterActions = (datasetType, language, shouldExist) => {
+  selectDataset(datasetType, language);
+  setDateRange(datasetType, language);
+
+  cy.getElementByTestId('docTable').get('tbody tr').should('have.length.above', 3); // To ensure it waits until a full table is loaded into the DOM, instead of a bug where table only has 1 hit.
+
+  const shouldText = shouldExist ? 'exist' : 'not.exist';
+  dataExplorer.getDocTableField(0, 0).within(() => {
+    cy.getElementByTestId('filterForValue').should(shouldText);
+    cy.getElementByTestId('filterOutValue').should(shouldText);
+  });
+
+  if (shouldExist) {
+    dataExplorer.verifyDocTableFilterAction(0, 'filterForValue', '10,000', '1', true);
+    dataExplorer.verifyDocTableFilterAction(0, 'filterOutValue', '10,000', '9,999', false);
+  }
+};
+const verifyExpandedTableFilterActions = (datasetType, language, isFilterButtonsEnabled) => {
+  // Check if the first expanded Doc Table Field's first row's Filter For, Filter Out and Exists Filter buttons are disabled.
+  const verifyFirstExpandedFieldFilterForFilterOutFilterExistsButtons = () => {
+    const shouldText = isFilterButtonsEnabled ? 'be.enabled' : 'be.disabled';
+    dataExplorer.getExpandedDocTableRow(0, 0).within(() => {
+      cy.getElementByTestId('addInclusiveFilterButton').should(shouldText);
+      cy.getElementByTestId('removeInclusiveFilterButton').should(shouldText);
+      cy.getElementByTestId('addExistsFilterButton').should(shouldText);
+    });
+  };
+
+  /**
+   * Check the Filter For or Out buttons in the expandedDocumentRowNumberth field in the expanded Document filters the correct value.
+   * @param {string} filterButton For or Out
+   * @param {number} docTableRowNumber Integer starts from 0 for the first row
+   * @param {number} expandedDocumentRowNumber Integer starts from 0 for the first row
+   * @param {string} expectedQueryHitsWithoutFilter expected number of hits in string after the filter is removed Note you should add commas when necessary e.g. 9,999
+   * @param {string} expectedQueryHitsAfterFilterApplied expected number of hits in string after the filter is applied. Note you should add commas when necessary e.g. 9,999
+   * @example verifyDocTableFirstExpandedFieldFirstRowFilterForButtonFiltersCorrectField('for', 0, 0, '10,000', '1');
```

Comment:
![Screenshot 2024-12-31 at 12 58 22 PM](https://github.com/user-attachments/assets/d18c1dd2-d779-400d-91ec-d44ab7b1873b)

Nice! I did not know about @example